proto/col_str.go

```diff
@@ -139,6 +127,11 @@ func (c *ColStr) DecodeColumn(r *Reader, rows int) error {
 			return errors.Wrapf(err, "row %d: read length", i)
 		}
 
+		if len(c.Buf)-p.End < binary.MaxVarintLen64 {
```

Comment:
struggling to understand the need for this change to the decoder. maybe a case of over-fitting for a test case?